packages/lib-utils/src/utils/url-sync.ts

```diff
@@ -0,0 +1,59 @@
+import type { History } from 'history';
+
+export const applyFiltersToURL = (
+  history: History,
+  keys: string[],
+  filterValues: Record<string, string>,
+) => {
+  const searchParams = new URLSearchParams(history.location.search);
+  keys.forEach((key: string) => {
+    searchParams.delete(key);
+    if (filterValues[key]) {
+      searchParams.set(key, filterValues[key]);
+    }
+  });
+
+  history.replace({
+    pathname: history.location.pathname,
+    search: searchParams.toString(),
+  });
+};
+
+export const syncFiltersWithURL = (
+  history: History,
+  keys: string[],
+  defaults: Record<string, string> = {},
+) => {
+  const searchParams = new URLSearchParams(history.location.search);
+
+  let filters: Record<string, string> = keys.reduce((acc, key) => {
+    const values = searchParams.getAll(key);
+    return {
+      ...acc,
+      [key]: values.length > 1 ? values : values[0],
```

Comment:
@fhlavac I wonder if we should update the logic below to handle the array in either scenario you're covering here - that way we are consistent and don't need as may `.length` checks?